Tasks/Common/azure-arm-rest/azure-arm-common.ts

```diff
@@ -49,11 +54,20 @@ export class ApplicationTokenCredentials {
         this.authorityUrl = authorityUrl;
         this.activeDirectoryResourceId = activeDirectoryResourceId;
         this.isAzureStackEnvironment = isAzureStackEnvironment;
+        this.scheme = scheme;
+        this.msiPort = msiPort;
     }
 
     public getToken(force?: boolean): Q.Promise<string> {
         if (!this.token_deferred || force) {
-            this.token_deferred = this.getAuthorizationToken();
+            if(this.scheme && this.scheme === "MSI")
```

Comment:
Can we have the scheme as Enum? (MSI , SPN)

Reply:
Done.